src/util/misc.ts

```diff
@@ -372,3 +368,50 @@ export function isBitmapImageMimeType(mimeType: string): boolean {
 export function isWebpCandidateImageMimeType(webp: boolean, content_type: string) {
   return webp && WEBP_CANDIDATE_IMAGE_MIME_TYPE.test(content_type)
 }
+
+/*
+ * Get best fitting MIME type from contentType or pathname
+ * Preference:
+ *   1. content-type if one of preferedDiscreteTypes
+ *   2. mime-type from extension if in preferedDiscreteTypes
+ *   3. any other content-type
+ *   4. any other mime-type from extension
+ *   5. null
+ */
+export function getMimeType(url: string, contentType?: string): string {
```

Comment:
The code looks good to me, but it's adapted to images. I don't think this is adapted for other kind of content (because then it tends to put the URL as prefered source, but for text content we rely on API and not sure what kind of result might come from `mime.lookup()`. Therefore I would recommend to rename to `getImageMimeType()`, I think this is more appropriate and might avoid calling at the wrong place.... Or I miss something important.